### PR TITLE
add hash reset

### PR DESCRIPTION
**Pull request by tudor-malene**
### Why this change is needed

properly reset the cached hash

### What changes were made as part of this PR

Please provide a high level list of the changes made

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: go/enclave/components/batch_executor.go [hunk 1]
```diff
@@ -110,6 +110,7 @@ func (executor *batchExecutor) ComputeBatch(context *BatchExecutionContext) (*Co
 
 	// we need to copy the batch to reset the internal hash cache
 	copyBatch := *batch
+	copyBatch.ResetHash()
```
Comment (BedrockSquirrel) [inline go/enclave/components/batch_executor.go:113]: I'd move this to below the mutation for clarity (line 115). That should be the pattern if we're using this Reset method, that every time properties are set the ResetHash gets called afterwards.

File: go/enclave/components/batch_executor.go [hunk 2]
```diff
@@ -110,6 +110,7 @@ func (executor *batchExecutor) ComputeBatch(context *BatchExecutionContext) (*Co
...
 	copyBatch.Header.Root = stateDB.IntermediateRoot(false)
 	copyBatch.Transactions = successfulTxs
```
Comment (BedrockSquirrel) [inline go/enclave/components/batch_executor.go:115]: Do we need to worry about `Header.TxHash` being out of sync with `copyBatch.Transactions` too?

**Review by BedrockSquirrel — APPROVED** [review]
I think this will fix the bug we saw in data dog, think we should make this harder to mess up in future though

File: go/enclave/components/batch_executor.go (hunk 2 above)
Comment (tudor-malene) [reply go/enclave/components/batch_executor.go:115]: I don't think so. But who knows

**Pull request merged**